prometheus_client/openmetrics/parser.py

```diff
@@ -83,95 +102,126 @@ def _parse_timestamp(timestamp):
             return ts
 
 
-def _parse_labels(it, text):
-    # The { has already been parsed.
-    state = 'startoflabelname'
-    labelname = []
-    labelvalue = []
+def _is_character_escaped(s, charpos):
+    num_bslashes = 0
+    while (charpos > num_bslashes and
```

Comment:
Shouldn't this comparison be against 0?

This is also going to be n^2 overall if there's many backslashes.

Reply:
it's n^2 in the worst case yes when we're escaping too many characters, I've mad a change by calling the function with smaller arguments `_is_character_escaped(value_substr[:i], i)` instead of `_is_character_escaped(value_substr, i)` 
Let me know if you're thinking of a possible better solution to optimize  👍